### ISSUE TITLE
Not support immersive-translate 0.8.0

### ISSUE THREAD

**Issue by cesaryuan** [JingMatrix/ChromeXt#93]
Thanks for your awesome project!

It seems that immersive-translate 0.8.0 is not supported. The floating button doesn't show up.

Other scripts works fine.

ChromeXt version: 3.6.0

**Comment by JingMatrix** [comment]
Did you install ChromeXt from GitHub Actions (not the releases)?
The related bug should have been fixed.

**Comment by cesaryuan** [comment]
> Did you install ChromeXt from GitHub Actions (not the releases)? The related bug should have been fixed.

Yes, I installed from actions [Inform users when page is sandboxed · JingMatrix/ChromeXt@ab6fb47](https://github.com/JingMatrix/ChromeXt/actions/runs/5885083941)

**Comment by cesaryuan** [comment]
1. open https://immersivetranslate.com/ to install script
2. open any page
3. button not show

https://github.com/JingMatrix/ChromeXt/assets/35998162/ca131d11-528b-4127-97c3-7eafeb136c1d

Xposed logs:

[LSPosed_2023-08-17T22_14_18.737322.zip](https://github.com/JingMatrix/ChromeXt/files/12370349/LSPosed_2023-08-17T22_14_18.737322.zip)

**Comment by JingMatrix** [comment]
Thanks for you video.
But I still cannot not reproduce it.

Can you check from the eruda Info panel whether this userscript is shown there?
If there are errors, eruda console might show something.

**Comment by cesaryuan** [comment]
> Can you check from the eruda Info panel whether this userscript is shown there?

Unfortunately, there is nothing in console panel. 😂

That's so weird, because that [kiss-translator](https://github.com/fishjar/kiss-translator)'s floating button can show up.

![image](https://github.com/JingMatrix/ChromeXt/assets/35998162/0e934698-019c-47f8-b8d7-490a25621bcd)

**Comment by cesaryuan** [comment]
> Maybe confusing, I meant eruda **Info** panel, not eruda **Console** panel.
> 
> By the way, things will be easier if you know how to inspect pages (it is taught in the introducing video).

Uncaught SyntaxError: Invalid regular expression: /[\u3041-\u3096\u309D-\u309F]|\uD82C[\uDC01-\uDD1F\uDD32\uDD50-\uDD52]|\uD83C\uDE00|[\u30A1-\u30FA\u30FD-\u30FF\u31F0-\u31FF\u32D0-\u32FE\u3300-\u3357\uFF66-\uFF6F\uFF71-\uFF9D]|\uD82B[\uDFF0-\uDFF3\uDFF5-\uDFFB\uDFFD\uDFFE]|\uD82C[\uDC00\uDD20-\uDD22\uDD55\uDD64-\uDD67]|[銗€-涠典竴-榫痌/g,jav:/[\uA980-\uA9CD\uA9D0-\uA9D9\uA9DE\uA9DF]/g: Range out of order in character class (at https://immersive-translate.owenyoung.com/:Immersive Translate:4100:3545)

![image](https://github.com/JingMatrix/ChromeXt/assets/35998162/8934ed50-e974-4a49-a35d-6e2671ee5d95)

**Comment by JingMatrix** [comment]
Okay, I know the problem now.
Your internet blocks a js lib (which is used to fix text encoding) for the code editor.

The solution for you is:
Download the script from __github release__ https://github.com/immersive-translate/immersive-translate/releases/tag/v0.8.0 and install it.

Becasuse the official website of immersive-translate has the problem of utf-8 encoding. (I am apparently not happy with the website).

**Comment by JingMatrix** [comment]
Make sure the Chinese characters in this page are shown correctly before you install it: 

![screenshot_2023-08-17T18:03:49](https://github.com/JingMatrix/ChromeXt/assets/24476093/8b915615-ba36-45aa-9cf2-4012240eacd8)

This issue is a duplicate of https://github.com/JingMatrix/ChromeXt/issues/54#issuecomment-1600477985

**Comment by cesaryuan** [comment]
> Make sure the Chinese characters in this page are shown correctly before you install it:安装前请确保该页面的汉字显示正确：
> 
> ![screenshot_2023-08-17T18:03:49](https://user-images.githubusercontent.com/24476093/261368281-8b915615-ba36-45aa-9cf2-4012240eacd8.png)
> 
> This issue is a duplicate of [#54 (comment)](https://github.com/JingMatrix/ChromeXt/issues/54#issuecomment-1600477985)这个问题是#54（评论）的重复

Thank you for your reply, I tried to download the script from Release, and found that there is still a problem. I checked the file with VSCode and found that the file is UTF-8 encoded. Do I need to change to another encoding?

![image](https://github.com/JingMatrix/ChromeXt/assets/35998162/d27ac8ca-89d7-49c9-8e72-922c21f034b3)

**Comment by cesaryuan** [comment]
Hi, I find the solution for it.

1. Download the script on PC
2. Using VSCode change the script  encoding to GB2312
3. Upload it to some server to get direct link (https://temp.getquicker.net/18359/45e8fc6a9c2749ab861bd6fd81360fec/immersive-translate%20%282%29.user.js)
4. Open the link in Chrome.

Now the script works fine!

**Comment by cesaryuan** [comment]
> running `console.log(document.characterSet)` in your console.

1. Run in "https://temp.getquicker.net/18359/fcd3ccb864704a6793909ced7e4fbd5b/immersive-translate%20%282%29.user.js", `gb18030`
2. Run in Github page, `UTF-8`

**Comment by cesaryuan** [comment]
> It is that your system has different encoding.

My system is Chinese Language 😂

**Comment by JingMatrix** [comment]
I see, some Chinese phones has `gb18030` as default encoding.
I might support to convert it in the future, if I find a js library to do the job. (It is quite tricky).

**Comment by cesaryuan** [comment]
> I see, some Chinese phones has `gb18030` as default encoding. I might support to convert it in the future, if I find a js library to do the job. (It is quite tricky)

Maybe we can request the `.user.js` in java level?

**Comment by JingMatrix** [comment]
If we do it with java, surely, it will give the correct encoding when you install it.
But somehow, I want to make it possible to edit the script in the page, which requires to do it with JS.

**Comment by cesaryuan** [comment]
> If we do it with java, surely, it will give the correct encoding when you install it. But somehow, I want to make it possible to edit the script in the page, which requires to do it with JS.

It is indeed a lot of trouble, but the conversion encoding scheme may not be completely converted back. Or an api can be provided, `https://example.com/convert-encoding?jsurl={original url}`, redirect the url of the user script

**Comment by JingMatrix** [comment]
An API endpoint will be very handy!

By the way, could you please copy and paste the wrongly decoded Chinese characters here? (I don't have a device to deconde wrongly).
So that I can test them with my JS codes.

**Comment by cesaryuan** [comment]
https://temp.getquicker.net/18359/868f34733d1540869e8073106a4049f4/immersive-translate.user.js

By the way, you can use vscode to reopen the utf-8 file with gb18030 encoding to get the wrongly decoded Chinese characters.

![image](https://github.com/JingMatrix/ChromeXt/assets/35998162/c2bce1fb-fa27-4e22-89b4-c50a03062bd3)

**Comment by JingMatrix** [comment]
Now gb18030 encoding can be fixed.
Please try and tell me if it is working.

**Comment by cesaryuan** [comment]
> Now gb18030 encoding can be fixed. Please try and tell me if it is working.

Thanks for your work. But unfortunately, it is still not working. As I can say, it is because that the converted text is wrong.

- Origin utf-8 encoding text `var test = /[㐀-䶵一-龯]/g` [test file](https://temp.getquicker.net/18359/54ddda2f0ded4c1dbe5c9e0e5fa89ef9/test.user.js)
- Converted text `
var test = /[㐢�-䶵一-龯]/g` 

![image](https://github.com/JingMatrix/ChromeXt/assets/35998162/9b1b71cf-a662-4060-a1c3-e1cd0323e1b7)

**Comment by JingMatrix** [comment]
I fixed the encoding algorithm.
It turns out this script happens to work now. Just a few characters in the metadata field not shown correctly.

**Comment by JingMatrix** [comment]
I found that there is a much easiler solution: Change you browser language to __Engligh__ and restart it.

English encoding are usually preserved (windows-1252) under unicode, unlike chines GBK encodings.

**Comment by cesaryuan** [comment]
> I found that there is a much easier solution: Change you browser language to **Engligh** and restart it.

Alse tried, not work haha 😂

**Comment by JingMatrix** [comment]
You need to ensure that your browser UI is in English, otherwise it won't takes effects.

**Comment by cesaryuan** [comment]
> You need to ensure that your browser UI is in English, otherwise it won't takes effects.

Yeah, I did change the browser UI to English but still not work haha. Maybe need to change whole android system not only the browser. Anyway, even if this is possible, it is still inconvenient for some people who are used to the Chinese system. 😄

**Comment by JingMatrix** [comment]
I found that Chrome did the correct encoding for most cases, and it behavior is a bit reasonable.
So there should be a function to check if the page has the correct encoding.
Now this is done, set commit  8da766b
